framework/i18n/GettextMessageSource.php

```diff
@@ -103,7 +103,11 @@ protected function loadFallbackMessages($category, $fallbackLanguage, $messages,
         $fallbackMessageFile = $this->getMessageFilePath($fallbackLanguage);
         $fallbackMessages = $this->loadMessagesFromFile($fallbackMessageFile, $category);
 
-        if ($messages === null && $fallbackMessages === null && $fallbackLanguage !== $this->sourceLanguage) {
+        if (
+            $messages === null && $fallbackMessages === null
+            && $fallbackLanguage !== $this->sourceLanguage
+            && $fallbackLanguage !== substr($this->sourceLanguage, 0, 2)
```

Comment:
We do not support something like `zh-Hant-CN` fully. I think current implementation is OK.
Since you did not update MessageSources in your PR yet, no conflicts should occur.